src/go/rdctl/pkg/plist/plist.go

```diff
@@ -25,46 +25,59 @@ import (
 const indentChange = "  "
 
 // convertToPListLines recursively reflects the supplied value into lines for a plist
-func convertToPListLines(v reflect.Value, indent string) ([]string, error) {
-	kind := v.Kind()
+func convertToPListLines(typedValue reflect.Value, actualValue reflect.Value, indent string) ([]string, error) {
+	kind := typedValue.Kind()
+	if actualValue.Kind() == reflect.Interface && typedValue.Kind() != reflect.Interface {
+		if actualValue.IsNil() {
+			return nil, nil
+		}
+		return convertToPListLines(typedValue, actualValue.Elem(), indent)
+	}
+	if actualValue.Kind() == reflect.Ptr {
+		return nil, fmt.Errorf("plist generation: got an unexpected pointer for value %v, typedValue %v\n", actualValue, typedValue)
+	}
 	switch kind {
 	case reflect.Struct:
-		numFields := v.NumField()
+		if actualValue.Kind() != reflect.Map {
+			return nil, fmt.Errorf("expecting actual kind for a typed struct to be a map, got %v", actualValue.Kind())
+		}
+		actualKeys := utils.SortKeys(actualValue.MapKeys())
+		numTypedFields := typedValue.NumField()
 		returnedLines := []string{indent + "<dict>"}
-		for i := 0; i < numFields; i++ {
-			fieldTag := v.Type().Field(i).Tag.Get("json")
+		// Typed fields are ordered according to options.ServerSettingsForJSON
+		// Actual fields are sorted by key (ignoring case), so walk the list of declared keys
+		// and take only the ones that appear in the actual instance. This gives a consistent,
+		// if not immediately obvious, order.
```

Comment:
It looks like we go through the whole sorting and stuff business to ensure case-insensitive matching for the keys? In that case, we should ensure have a test case that catches that.  If that's unintentional, we should instead skip sorting (and iterating through) the map keys.

Reply:
There are bats tests that verify the output-order. I added comments, and reordered two fields in the input, to highlight this.